src/firefly/js/templates/fireflyslate/FireflySlateManager.js

```diff
@@ -0,0 +1,216 @@
+            TABLE_REMOVE, TABLE_LOADED, TBL_RESULTS_ADDED, TBL_RESULTS_ACTIVE,
+            CHART_ADD, CHART_REMOVE,
+            SHOW_DROPDOWN, SET_LAYOUT_MODE, ENABLE_SPECIAL_VIEWER
+        ]);
+
+        /**
+         * This is the current state of the layout store.  Action handlers should return newLayoutInfo if state changes
+         * If state has changed, it will be dispatched into the flux.
+         * @type {LayoutInfo}
+         * @prop {boolean}  layoutInfo.showTables  show tables panel
+         * @prop {boolean}  layoutInfo.showXyPlots show charts panel
+         * @prop {boolean}  layoutInfo.showImages  show images panel
+         * @prop {string}   layoutInfo.searchDesc  optional string describing search criteria used to generate this result.
+         * @prop {boolean}  layoutInfo.autoExpand  this is true when manager think it should be expanded, ie. single view
+         * @prop {Object}   layoutInfo.images      images specific states
+         * @prop {string}   layoutInfo.images.metaTableId  tbl_id of the image meta table
+         * @prop {string}   layoutInfo.images.selectedTab  selected tab of the images tabpanel
+         * @prop {string}   layoutInfo.images.showCoverage  show images coverage tab
+         * @prop {string}   layoutInfo.images.showFits  show images fits data tab
+         * @prop {string}   layoutInfo.images.showMeta  show images image metea tab
+         * @prop {string}   layoutInfo.images.coverageLockedOn
+         */
+        var layoutInfo = getLayouInfo();
+        var newLayoutInfo = layoutInfo;
+
+        switch (action.type) {
+            case ImagePlotCntlr.PLOT_IMAGE_START:
+            case ImagePlotCntlr.PLOT_IMAGE :
+            case REPLACE_VIEWER_ITEMS:
+                newLayoutInfo = handleNewImage(newLayoutInfo, action);
+                break;
+            case ImagePlotCntlr.DELETE_PLOT_VIEW:
+                newLayoutInfo = handlePlotDelete(newLayoutInfo, action);
+                break;
+            // case TABLE_LOADED:
+            case TBL_RESULTS_ADDED:
+                newLayoutInfo = handleNewTable(newLayoutInfo, action);
+                break;
+            case TABLE_REMOVE:
+                newLayoutInfo = handleTableDelete(newLayoutInfo, action);
+                break;
+            case CHART_ADD:
+                newLayoutInfo = handleNewChart(newLayoutInfo, action);
+                break;
+            case CHART_REMOVE:
+                newLayoutInfo = handleChartDelete(newLayoutInfo, action);
+                break;
+            case ENABLE_SPECIAL_VIEWER:
+                const startCellId= startSepcialViewerSaga(action, alreadyStartSagas);
+                if (startCellId) alreadyStartSagas= [...alreadyStartSagas,startCellId];
+                break;
+        }
+
+
+        if (newLayoutInfo !== layoutInfo) {
+            dispatchUpdateLayoutInfo(newLayoutInfo);
+        }
+    }
+}
+
+
+function startSepcialViewerSaga(action, alreadyStarted) {
```

Comment:
I don't think it needs a stop since the saga's don't really take that much resources. I also don't think the basic use case will be to be bringing all different types and removing them.  The user will get a layout he likes for his work put up the data